ldap/client/api/src/main/java/org/apache/directory/ldap/client/api/LdapConnectionPool.java

```diff
@@ -123,6 +123,10 @@ public LdapConnection getConnection() throws LdapException
         try
         {
             connection = super.borrowObject();
+            if ( connection instanceof PooledLdapConnection )
+            {
+                ( ( PooledLdapConnection ) connection ).setConnectionPool( this );
+            }
```

Comment:
I ran into a chicken and egg problem with the pooled connection, so this was the work around...

Ideally I'd want to pass the pool in the constructor of the `LdapConnectionFactory`, but that itself is passed indirectly into the `AbstractPoolableLdapConnectionFactory`

An alternative, _might_ be to have the `LdapConnectionPool`'s constructor inspect the type `PooledObjectFactory`, if it's a `AbstractPoolableLdapConnectionFactory`, the connection factory could be wrapped 🤔
Similar solution, different location 🤷